### ISSUE TITLE
Apply plugin in multi project gradle configure error

### ISSUE THREAD

**Issue by DaiYuANg** [beryx/badass-jlink-plugin#265]
I have a root project include subproject like

- root project
	- subproject 1
	- subproject 2

When my root project applies this plugin and implementation my subproject 1 and 2 that will show this error
```
Cause: org.gradle.api.internal.initialization.DefaultClassLoaderScope@xxxx must be locked before it can be used to compute a classpath!
```

**Comment by DaiYuANg** [comment]
This error at 
PrepareMergedJarsDirTask.groovy:70
```
 PrepareMergedJarsDirTask() {
        description = 'Merges all non-modularized jars into a single module'
        project.afterEvaluate {
            def projects = Util.getAllDependentProjects(project) + project
            def jarTasks = projects*.getTasksByName('jar', true).flatten() as Task[]
            dependsOn(jarTasks)
        }
    }
```

**Comment by xzel23** [comment]
Included in 3.1.0-rc-1. It would be nice if you could provide a PR that includes a test case anyway. I use multiple modules in some of my own projects and they work with your changes, but these also worked before including the change, probably because of another project structure.
